linters/import/main.go

```diff
@@ -0,0 +1,438 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package main
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"go/ast"
+	"go/build"
+	"go/parser"
+	"go/token"
+	"go/types"
+	"log"
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/kisielk/gotool"
+	"golang.org/x/tools/go/loader"
+)
+
+const (
+	standard = "standard"
+	external = "ext"
+)
+
+var (
+	errMultipleImport       = errors.New("more than one import group found")
+	errOutOfOrder           = errors.New("import is out of order")
+	errImportMatchedAlready = errors.New("import already matched pattern")
+	errGroupMachedAlready   = errors.New("import group already matches previously seen pattern")
+	errNoMatch              = errors.New("import does not match any of the provided patterns")
+	errTooManyImports       = errors.New("number of import groups exceeds number of patterns. this usually means a previous or future import is out of order or you didn't specify the correct patterns")
+)
+
+type lintError struct {
+	fileName    string
+	importName  string
+	line        int
+	patternSeen string
+	err         error
+}
+
+type lintErrors []lintError
+
+// NB(prateek): http://goast.yuroyoro.net/ is enormously helpful.
+
+func main() {
+	tags := flag.String("tags", "", "List of build tags to take into account when linting.")
+	skipVendor := flag.Bool("skip-vendor", true, "Skip vendor directors.")
+	rawPatterns := flag.String("patterns", "", "Specify the patterns of each group in order. If checking for Go standard imports write `standard`, if checking for a wildard group write `all`.")
```

Comment:
Better to have a default of `STANDARD_LIBRARY` + `THIRD_PARTY` than the empty string.